Add simplified wrappers to avoid misuse

mostynb opened the pull request:
Added `NewDecoderPoolWrapper` and `NewEncoderPoolWrapper` which return
sync.Pool wrappers which don't need client-side type assertions.

Implements #4.

coder543 commented on decoderpool.go:
```diff
@@ -96,3 +96,32 @@ func NewDecoderPool(options ...zstd.DOption) *sync.Pool {
+// Get returns a *DecoderWrapper that has been Reset to use r.
+func (d *DecoderPoolWrapper) Get(r io.Reader) *DecoderWrapper {
+	dw := d.pool.Get().(*DecoderWrapper)
+	_ = dw.Reset(r)
```
Hiding this error doesn't feel super great... it looks like the main reason it would return an error is if the underlying decoder was in a permanently broken state. Maybe this function should bubble that error up to the caller? Or maybe it should keep fetching decoders from the pool and attempting to reset them until it finds one that doesn't error? I'm not sure what your thoughts are on this, but hiding the error could lead the caller being given a broken decoder and not knowing it.

mostynb replied:
In v1.13.6, Decoder.Reset only returns a non-nil error if the Decoder previously had its Close method called:
https://github.com/klauspost/compress/blob/v1.13.6/zstd/decoder.go#L158

But our `DecoderWrapper.Close()` intentionally doesn't call `Decoder.Close()`, because that would render it unusuable- instead it resets it with a nil reader and adds it back to the pool:
https://github.com/mostynb/zstdpool-syncpool/blob/0d9913b62da43f38830e3b8d82aaa0b721b3dd69/decoderpool.go#L46

So I think the only way this would cause trouble is if the client code type-asserted this to a zstd.Decoder and then closed it. Maybe it's worth panicking if we catch an error here?

coder543 replied:
Ah, that's a good point, and I agree. I think a panic is probably the right thing to do here.

mostynb committed:
Add simplified wrappers to avoid misuse

Added NewDecoderPoolWrapper and NewEncoderPoolWrapper which return
sync.Pool wrappers which don't need client-side type assertions.

Implements #4.
